app/concepts/tribunal_commerce/partial_stock_unit/operation/load.rb

```diff
@@ -2,6 +2,87 @@ module TribunalCommerce
+
+        step :read_files_metadata
+          fail :log_filenames_parsing_error, fail_fast: true
+
+        step :supersede_dossiers_entreprise_from_pm
+        step :import_personnes_morales
+        step :supersede_dossiers_entreprise_from_pp
+        step :import_personnes_physiques
+        step :import_representants
+        step :import_etablissements
+        step :import_observations
+
+        def prepare_zip_extraction(ctx, daily_update_unit:, **)
+          ctx[:path] = daily_update_unit.files_path
+        end
+
+        def log_zip_extraction_error(ctx, daily_update_unit:, error:, logger:, **)
+          logger.error("An error happened while trying to extract unit archive #{daily_update_unit.files_path} : #{error}")
+        end
+
+        def read_files_metadata(ctx, extracted_files:, **)
+          ctx[:files_args] = parse_stock_filename(extracted_files)
+        rescue TribunalCommerce::Helper::DataFile::UnexpectedFilename => e
+          ctx[:error] = e.message
+          false
+        end
+
+        def log_filenames_parsing_error(ctx, logger:, error:, **)
+          logger.error("An error occured while parsing unit's files : #{error}")
+        end
+
+        def supersede_dossiers_entreprise_from_pm(ctx, files_args:, file_importer:, **)
+          pm_file_path = extract_file_path(files_args, 'PM')
+          file_importer.supersede_dossiers_entreprise_from_pm(pm_file_path)
+        end
+
+        def supersede_dossiers_entreprise_from_pp(ctx, files_args:, file_importer:, **)
+          pp_file_path = extract_file_path(files_args, 'PP')
+          file_importer.supersede_dossiers_entreprise_from_pp(pp_file_path)
+        end
+
+        def import_personnes_morales(ctx, files_args:, file_importer:, **)
+          pm_file_path = extract_file_path(files_args, 'PM')
+          file_importer.import_personnes_morales(pm_file_path)
+        end
+
+        def import_personnes_physiques(ctx, files_args:, file_importer:, **)
+          pp_file_path = extract_file_path(files_args, 'PP')
+          file_importer.import_personnes_physiques(pp_file_path)
+        end
+
+        def import_representants(ctx, files_args:, file_importer:, **)
+          rep_file_path = extract_file_path(files_args, 'rep')
+          file_importer.import_representants(rep_file_path)
+        end
+
+        def import_etablissements(ctx, files_args:, file_importer:, **)
+          ets_file_path = extract_file_path(files_args, 'ets')
+          file_importer.import_etablissements(ets_file_path)
+        end
+
+        def import_observations(ctx, files_args:, file_importer:, **)
+          obs_file_path = extract_file_path(files_args, 'obs')
+          file_importer.import_observations(obs_file_path)
+        end
+
+        def extract_file_path(files_args, label)
+          files_args
+            .yield_self { |it| it.select { |file_args| file_args[:label] == label } }
+            .yield_self { |it| it.first }
```

Comment:
`select` + `first` = `find`

Reply:
Ouais sauf qu'ici j'ai un tableau de Hash et pas un ActiveRecord::Relation

Reply:
si si ça marche c'est une méthode de `Array`

Reply:
Ok il y a bien une méthode `find` pour les Enumerator (et donc les tableaux), which is nice !